src/Microsoft.Azure.WebJobs.Host/Blobs/BindableBlobPath.cs

```diff
@@ -1,45 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-
-using System;
-using System.Linq;
-using Microsoft.Azure.WebJobs.Host.Bindings.Path;
-
-namespace Microsoft.Azure.WebJobs.Host.Blobs
-{
-    internal static class BindableBlobPath
-    {
-        public static IBindableBlobPath Create(string pattern, bool isContainerBinding = false)
-        {
-            BlobPath parsedPattern = null;
-            // try blob/container pattern from url or string
-            if (!BlobPath.TryParseAbsUrl(pattern, out parsedPattern) &&
-                !BlobPath.TryParse(pattern, isContainerBinding, out parsedPattern))
-            {
-                // if failed => not blob/container pattern
-                // we try pattern url
-                BindingTemplate urlTemplate = BindingTemplate.FromString(pattern);
-                if (urlTemplate.ParameterNames.Count() == 1)
-                {
-                    return new ParameterizedBlobUrl(urlTemplate);
-                }
-                throw new FormatException($"Invalid blob path '{pattern}'. Paths must be in the format 'container/blob' or 'blob Url'.");
```

Comment:
Where does this happen now?

Reply:
I'll create a BlobName validation data annotation attribute.  This goes through attribute cloner, so the parameterized and non-parameterized cases get unified and it will catch the case below too.

---
In reply to: [150936782](https://github.com/Azure/azure-webjobs-sdk/pull/1422#discussion_r150936782) [](ancestors = 150936782)